crates/nu-command/src/strings/str_/join.rs

```diff
@@ -14,6 +16,11 @@ impl Command for StrJoin {
                 (Type::List(Box::new(Type::Any)), Type::String),
                 (Type::String, Type::String),
             ])
+            .switch(
+                "stream",
+                "Output the result as a raw stream, instead of collecting to a string.",
+                Some('s'),
+            )
```

Comment:
I assume this could break the static type checking here.

There may be commands that don't declare `Type::String` as acceptable but would happily work on a raw stream (Limitation 1: we do not have a `Type` for that).

Furthermore our type propagation logic can only assume a different output type if the input type is distinct. Internal behavior or added flags can not affect the static output type. (Limitation 2: it is just propagating not inferring, Limation 3: signature at the command level not at the usage level.)

Reply:
Yeah, that makes sense. 

I've been curious about `Type::ListStream` for a while since it doesn't seem to get much use. Would a `Type::ListStream` to hypothetical `Type::RawStream` work instead or would that be too much of a gotcha?